### PR TITLE
Py speedup1

### LINKED ISSUE #1839: Small python improvements
- [x] return numpy arrays for some properties instead of Vec
- [x] query the global graph for the edge id

### PR DESCRIPTION
**Pull request by fabianmurariu**
### What changes were proposed in this pull request?
Add numpy to python properties when we can

### Why are the changes needed?
because rust -> python is sometimes slow

### Issues

closes #1839

### CODE REVIEW

**Comment by fabianmurariu** [comment]
I'm a bit puzzled about the benchmarks, this is what I'm getting
```
lotr_graph_window_50_layered/iterate nodes
                        time:   [7.1942 µs 7.2791 µs 7.3649 µs]
                        change: [-28.904% -26.641% -24.581%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild

  2024-10-27T20:20:18.813714Z  INFO raphtory_benchmark::common: Num layers 5, node count: 139, edge_count: 701
    at raphtory-benchmark/src/common/mod.rs:275

lotr_graph_persistent_window_50_layered/iterate nodes
                        time:   [8.3778 µs 8.4859 µs 8.5862 µs]
                        change: [-32.056% -30.450% -28.763%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild

graph: Graph(num_nodes=139, num_edges=701)
graph: Graph(num_nodes=139, num_edges=701)
murariuf@fedora ~/S/raphtory (py_speedup1)> cargo bench --bench base --bench base -p raphtory-benchmark "iterate nodes" -- --baseline py_speedup1 --load-baseline master
```

File: raphtory/src/db/graph/node.rs [hunk 1]
```diff
@@ -74,7 +74,7 @@ impl<'graph, G, GH: GraphViewOps<'graph> + Debug> fmt::Debug for NodeView<G, GH>
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_struct("NodeView")
             .field("node", &self.node)
-            .field("graph", &self.graph)
+            .field("graph", &self.graph as &dyn Debug)
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/node.rs:77]: why is this needed now?

File: raphtory/src/python/graph/edge.rs [hunk 2]
```diff
@@ -176,8 +177,18 @@ impl PyEdge {
...
+        let history = self.edge.history();
+        history.into_pyarray(py).to_owned()
+    }
+
+    /// Returns the number of times an edge is added or change to an edge is made.
+    ///
+    /// Returns:
+    ///    int: The number of times an edge is added or change to an edge is made.
+    ///
+    pub fn history_counts<'py>(&self, py: Python<'py>) -> u64 {
```
Comment (ljeub-pometry) [inline raphtory/src/python/graph/edge.rs:190]: you can just return `usize`?

File: raphtory/src/python/graph/properties/temporal_props.rs [hunk 3]
```diff
@@ -258,7 +258,7 @@ impl PyTemporalProp {
 
     /// Iterate over `items`
     pub fn __iter__(&self) -> PyGenericIterator {
-        self.prop.iter().into()
+        self.prop.histories().collect::<Vec<_>>().into_iter().into()
```
Comment (ljeub-pometry) [inline raphtory/src/python/graph/properties/temporal_props.rs:261]: could use the lifetime version

File: raphtory/src/db/graph/node.rs (hunk 1 above)
Comment (fabianmurariu) [reply raphtory/src/db/graph/node.rs:77]: rust-analyzer

**Review by ljeub-pometry — APPROVED** [review]
Looks good to merge
